src/checker/CidChecker.ts

```diff
@@ -0,0 +1,210 @@
+      FROM miners
+               LEFT OUTER JOIN active_miners ON miners.provider = active_miners.miner_id
+      ORDER BY total_deal_size DESC`
+
+  private static readonly ReplicaDistributionQuery = `
+      WITH replicas AS (SELECT COUNT(*) AS num_of_replicas, SUM(piece_size) AS total_deal_size, piece_cid
+                        FROM current_state,
+                             client_mapping
+                        WHERE client_address = $1
+                          AND current_state.client = client_mapping.client
+                          AND verified_deal = true
+                          AND slash_epoch < 0
+                          AND (sector_start_epoch > 0 AND sector_start_epoch < $2)
+                          AND end_epoch > $2
+                        GROUP BY piece_cid)
+      SELECT SUM(total_deal_size) AS total_deal_size, num_of_replicas::INT
+      FROM replicas
+      GROUP BY num_of_replicas
+      ORDER BY num_of_replicas ASC`
+
+  private static readonly CidSharingQuery = `
+      WITH cids AS (SELECT DISTINCT piece_cid
+                    FROM current_state,
+                         client_mapping
+                    WHERE client_address = $1
+                      AND current_state.client = client_mapping.client
+                      AND verified_deal = true)
+      SELECT SUM(piece_size)                              AS total_deal_size,
+             COUNT(DISTINCT current_state.piece_cid)::INT AS unique_cid_count,
+             client_address                               as other_client_address
+      FROM cids,
+           current_state,
+           client_mapping
+      WHERE cids.piece_cid = current_state.piece_cid
+        AND current_state.client = client_mapping.client
+        AND client_address != $1
+      GROUP BY client_address
+      ORDER BY total_deal_size DESC`
+
+  public constructor (
+    private readonly sql: Pool,
+    private readonly logger: Logger) {
+  }
+
+  private static getProjectNameFromTitle (titleStr: string): string {
+    let title = titleStr.replace(/\[DataCap\s*Application\]/i, '')
+    const splitted = title.trim().split('-')
+    if (splitted.length >= 2) {
+      title = splitted.slice(1).join('-')
+    }
+    title = title.trim()
+    if (title.startsWith('<') && title.endsWith('>')) {
+      title = title.substring(1, title.length - 1)
+    }
+    return title
+  }
+
+  private static getApplicationInfo (issue: Issue): ApplicationInfo {
+    const parseResult = parseIssue(issue.body ?? '')
+    if (!parseResult.correct) {
+      throw new Error(`Invalid issue body.\n  errorMessage: ${parseResult.errorMessage}\n  errorDetails: ${parseResult.errorDetails}`)
+    }
+
+    return {
+      clientAddress: parseResult.address,
+      organizationName: parseResult.name,
+      projectName: CidChecker.getProjectNameFromTitle(issue.title)
+    }
+  }
+
+  private static getCurrentEpoch (): number {
+    return Math.floor((Date.now() / 1000 - 1598306400) / 30)
```

Comment:
can you confirm that 1598306400 is the Filecoin Genesis epoch? 

was confused seeing this in different places.

Reply:
Yes, it is